JWBestPracticeApps/FeedTableViewController/FeedTableViewController/Feed/JWV2Playlist.swift

```diff
@@ -0,0 +1,1023 @@
+
+/// This file was initially generated from JSON Schema using quicktype, MODIFY AT YOUR OWN RISK.
+/// To parse the JSON:
+///
+/// ```
+/// let jwDecoder = JSONDecoder()
+/// jwDecoder.keyDecodingStrategy  = .convertFromSnakeCase // so auto-coding keys work (except for "description")
+/// jwDecoder.dateDecodingStrategy = .secondsSince1970     // so Date decoding works
+///
+/// let jWV2Playlist = try? decoder.decode(JWManualPlaylist.self, from: jsonData)
+struct JWV2Playlist: Decodable {
+    let title: String?
+    let description: String?
+    let feedInstanceId: UUID?
+    let feedid: String?
+    let kind: PlaylistType?
+    let links: Links?
+    let playlist: [PlaylistItem]?
+    
+    static let jwDecoder: JSONDecoder = {
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy  = .convertFromSnakeCase
+        decoder.dateDecodingStrategy = .secondsSince1970
+        return decoder
+    }()
+    
+    var jwDecoder: JSONDecoder { Self.jwDecoder }
+    
+    // For demo
+    static let exampleManualPlaylist: JWV2Playlist = {
+        try! jwDecoder.decode(Self.self, from: exampleManualPlaylistResponse.data(using: .utf8)!)
+    }()
+}
+
+enum PlaylistType: String, Decodable {
+    case MANUAL
+    case DYNAMIC
+}
+
+// MARK: - Links
+struct Links: Decodable {
+    let first: URL?
+    let last: URL?
+}
+
+// MARK: - PlaylistItem
+struct PlaylistItem: Decodable {
+    let description: String?
+    let duration: Int?
+    let feedid: String?
+    let image: URL?
+    let images: [Image]?
+    let link: URL?
+    let mediaid: String?
+    let pubdate: Date?
+    let sources: [Source]?
+    let title: String?
+    let tracks: [Track]?
+    let tags: String?
+    var tagsArray: [String] {
+        tags?
+            .split(separator: ",")
+            .compactMap {String($0)}
+        ?? []
+    }
+    // TODO: Implement this
+//    let variations: Any?
+}
+
+// MARK: - Image
+struct Image: Decodable {
+    let src: URL?
+    let type: SourceType?
+    let width: Int?
+}
+
+// MARK: - Source
+struct Source: Decodable {
+    let bitrate: Int?
+    let file: URL?
+    let filesize: Int?
+    let framerate: Int?
+    let height: Int?
+    let label: String?
+    let type: SourceType?
+    let width: Int?
+}
+
+// It's a shame that UTTypeReference is not Decodable
+enum SourceType: String, Decodable {
+    case applicationVndAppleMpegurl = "application/vnd.apple.mpegurl"
+    case audioMp4  = "audio/mp4"
+    case videoMp4  = "video/mp4"
+    case imageJpeg = "image/jpeg"
+    case imagePng  = "image/png"
+}
+
+// MARK: - Track
+struct Track: Decodable {
+    let file: URL?
+    let kind: TrackType?
+    let label: String?
+}
+
+enum TrackType: String, Decodable {
+    case thumbnails = "thumbnails"
+    case captions   = "captions"
+}
+
+
+// MARK: - exampleManualPlaylistResponse
+
+/// Response from a call to the Delivery API, `https://cdn.jwplayer.com/v2/playlists/d7sk9orq`
+fileprivate let exampleManualPlaylistResponse = """
```

Comment:
We do not know where a developer is getting their playlist from, so the format below is not known to them. Perhaps you should be using an array of configs instead?

A great rule of thumb for BPA is to keep it as simple as possible. Having server responses and decoding/serialization of data is not simple or easily understood.